programs/ssl/ssl_client2.c

```diff
@@ -135,9 +140,23 @@ int main( void )
 #endif /* MBEDTLS_X509_CRT_PARSE_C */
 
 #if defined(MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED)
-#define USAGE_PSK                                                   \
+#define USAGE_PSK_RAW                                               \
     "    psk=%%s              default: \"\" (in hex, without 0x)\n" \
     "    psk_identity=%%s     default: \"Client_identity\"\n"
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#define USAGE_PSK_SLOT                          \
+    "    psk_slot=%%d         default: 0\n"     \
```

Comment:
As discussed in the PR that adds something similar for ECDSA keys, I think it would be better to call the option `psk_opaque` and make it a boolean. That way when PSA changed the way keys are accessed, we only need to change the option's implementation, not its interface.

Reply:
Yes! At the time of writing the PR, I didn't expect a slot allocation mechanism, nor thought of our intermediate approach of searching through the slot list ourselves. I will adapt the PR to follow that approach, and consequently rename `psk_slot` to `psk_opaque`.